05_Datathon/diabetes_prediction_example.Rmd

```diff
@@ -0,0 +1,336 @@
+
+# options(gargle_oauth_email = TRUE)
+options(gargle_verbosity = "info")
+
+```
+
+```{r}
+
+# Checking that all required packages are installed
+if(!require("tidyverse")) install.packages("tidyverse")
+if(!require("tableone")) install.packages("tableone")
+if(!require("caret")) install.packages("caret")
+if(!require("dplyr")) install.packages("dplyr")
+if(!require("rpart")) install.packages("rpart")
+if(!require("rpart.plot")) install.packages("rpart.plot")
+if(!require("ROCR")) install.packages("ROCR")
+if(!require("knitr")) install.packages("knitr")
+if(!require("randomForest")) install.packages("randomForest")
+
+# List of packages to load
+packages <- c("tidyverse", "tableone", "caret","dplyr","ROCR", "knitr","randomForest")
+
+# Load packages
+lapply(packages, FUN = function(X) {
+    do.call("require", list(X)) 
+})
+```
+
+
+### 1.2 Load in the Dataset
+
+Now we can load in our data and start to work with it.
+
+```{r}
+# Read in data (might take a few seconds)
+gossis <- read_csv("gossis_post_process-eicu-only-wids.csv")
+```
+
+```{r}
+# Check dimensions of data
+# Output format:  rows  by columns
+dim(gossis)
+``` 
+
+We can see our data has `r dim(gossis)[1]` entries and `r dim(gossis)[2]` Variables. Let's see what our data actually looks like!
+
+```{r}
+head(gossis[,1:7]) %>% kable(caption = "Gossis (1st seven variables)")
+```
+
+Next, we will select the variables we want to examine, some of which may end up in our final model. Here, we choose to include the following: `bmi`, `age`, `ethnicity`, `gender`, `icu_type`, and our target variable: `diabetes_mellitus`.
+
+```{r}
+#  Select subset of variables from original data
+reduced_data <- gossis %>% select(., 
+                                  bmi, 
+                                  age,
+                                  gender,
+                                  ethnicity,
+                                  icu_type,
+                                  glucose_apache,
+                                  diabetes_mellitus)
+
+head(reduced_data) %>% kable()
+```
+
+## 2. Preprocessing and Missing Data
+
+We have to consider the NA values in our data. In a research project, it is critical to investigate the missing data by searching for patterns and replacing missing values (imputation) to see if this affects conclusions. For the simplicity of this demo, we will remove all rows containing missing values.
+
+However, in your project, feel free to experiment with different imputation techniques!
+
+```{r data_types, echo=TRUE}
+# Check to see number of missing values in each column
+colSums(is.na(reduced_data))
+
+# Remove rows with NA for age, bmi, ethnicity, gender, or diabetes_mellitus
+reduced_data <- drop_na(reduced_data)
+
+# We can see there are no NA values in our data
+colSums(is.na(reduced_data))
+```
+
+Next we will encode our categorical variables. Encoding is the process of reshaping and binarizing categorical data to better suit machine learning models.
+
+```{r}
+
```

Comment:
technically all transformations like this should be done on the training set.  i don't think it will matter for this particular example, but it's a bad habit to teach.